singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -684,8 +691,15 @@ class PTESolverRhoT : public mix_impl::PTESolverBase<EOSIndexer, RealIndexer> {
       const Real rho_min =
           std::max(eos[m].RhoPmin(Tnorm * Tequil), eos[m].RhoPmin(Tnorm * Tnew));
       const Real alpha_max = robust::ratio(rhobar[m], rho_min);
+      if (alpha_max < vfrac[m]) {
+        // Despite our best efforts, we're already in the unstable regime (i.e.
+        // dPdV_T > 0) so we would actually want to *increase* the step instead
+        // of decreasing it. As a result, this code doesn't work as intended and
+        // should be skipped.
+        continue;
+      }
       if (scale * dx[m] > 0.5 * (alpha_max - vfrac[m])) {
-        scale = robust::ratio(0.5 * alpha_max - vfrac[m], dx[m]);
+        scale = robust::ratio(0.5 * (alpha_max - vfrac[m]), dx[m]);
```

Comment:
There were two bugs here:
1. If the volume fraction is already greater than this maximum and the current step isn't large enough to bring it back, we need a _larger_ step, not a smaller step. This goes against the logic of this function in limiting the step and could undo other limits. As a result, I just give up on this limit if we're already beyond it.
2. There was a missing set of parentheses in the eventual scale update.

Reply:
Nice catch!

Reply:
nice find